integration/internal/process.go

```diff
@@ -89,14 +96,17 @@ func DoDefaulting(
 func (ps *ProcessState) Start(stdout, stderr io.Writer) (err error) {
 	command := exec.Command(ps.Path, ps.Args...)
 
-	startDetectStream := gbytes.NewBuffer()
-	detectedStart := startDetectStream.Detect(ps.StartMessage)
+	ready := make(chan bool)
 	timedOut := time.After(ps.StartTimeout)
 
-	if stderr == nil {
-		stderr = startDetectStream
+	if ps.HealthCheckEndpoint != "" {
+		healthCheckURL := ps.URL
+		healthCheckURL.Path = ps.HealthCheckEndpoint
+		go pollURLUntilOK(healthCheckURL, ready)
```

Comment:
If the start timeout is reached, this goroutine should be terminated.

Reply:
Right, thanks! 👍